src/main/java/com/beerjournal/breweriana/user/UserController.java

```diff
@@ -33,13 +34,28 @@
 
     @PutMapping("{userId}")
     ResponseEntity<UserDto> modifyUserWithId(@PathVariable(value = "userId") String userId,
-                                           @RequestBody @Validated UserDto user) {
+                                             @RequestBody @Validated UserDetailsDto user) {
         return new ResponseEntity<>(userService.modifyUserWithId(userId, user), HttpStatus.OK);
     }
 
     @DeleteMapping("{userId}")
-    ResponseEntity<UserDto> deleteUserWithId(@PathVariable(value = "userId") String userId) {
-        return new ResponseEntity<>(userService.deleteUserWithId(userId), HttpStatus.OK);
+    ResponseEntity<UserDto> deleteUserWithId(@PathVariable(value = "userId") String userId,
+                                             @RequestBody @Validated UserDeleteDto user) {
+        return new ResponseEntity<>(userService.deleteUserWithId(userId, user), HttpStatus.OK);
+    }
+
+    @PostMapping("{userId}/password")
```

Comment:
Maybe we should extract endpoints related to changing user account to /api/account ? What do do you think?